### PR TITLE
Ports Builder-based services to SpanStore and deletes Index and Storage

### LINKED ISSUE #483: Lots of code duplication?
Looks like zipkin-common has lots of code duplication. Or is it just me not understanding the code?

E.g.
https://github.com/openzipkin/zipkin/blob/master/zipkin-common/src/main/scala/com/twitter/zipkin/storage/Storage.scala
has many of the same methods as
https://github.com/openzipkin/zipkin/blob/master/zipkin-common/src/main/scala/com/twitter/zipkin/storage/SpanStore.scala

I'm wondering if it's https://github.com/openzipkin/zipkin/pull/321 that was merged with new interfaces, without removing the old ones. @sprsquish @bmdhacks @franklinhu could that be right?

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This completes code de-duplication by removing all references to Index
and Storage, after porting zipkin-collector-service and
zipkin-query-service to use SpanStore-based implementations.

closes #483

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
tested using `bin/collector`, `bin/query`, `bin/tracegen`, and `bin/web` on dev cassandra and redis

File: zipkin-query-service/src/main/scala/com/twitter/zipkin/query/Main.scala [hunk 1]
```diff
@@ -35,11 +36,15 @@ object Main {
...
-      val server = builder.apply().apply(runtime)
-      server.start()
-      ServiceTracker.register(server)
+      val query = builder.apply().apply(runtime)
+      Await.ready(query)
+      ServiceTracker.register(new Service() {
+        override def shutdown() = Await.ready(query.close())
+
+        override def start() = {}
+      })
```
Comment (jamescway) [inline zipkin-query-service/src/main/scala/com/twitter/zipkin/query/Main.scala:47]: I don't have a lot of experience with starting ostrich services, so I hope this is the accept way of doing it.  Perhaps a bit of shoehorning legacy?

**Comment by jamescway** [comment]
Done have a ton of context, but everything lgtm except that one thing, which I'm assuming is necessary.
:+1:  nice deletes :100:

File: zipkin-query-service/src/main/scala/com/twitter/zipkin/query/Main.scala (hunk 1 above)
Comment (codefromthecrypt) [inline zipkin-query-service/src/main/scala/com/twitter/zipkin/query/Main.scala:47]: Yeah, it is shoe-horning. Migrating zipkin-collector-server and zipkin-query-server off ostrich is yet another todo :) Where to, is yet another question! finatra, finch etc.

**Pull request merged**